eclipse/src/saros/ui/command_handlers/ShareResourcesHandler.java

```diff
@@ -35,12 +38,27 @@
  *
  * <p>Notice that this is done via the {@link saros.ui.wizards.StartSessionWizard}.
  */
-public class ShareResourcesHandler extends AbstractHandler {
+public class ShareResourcesHandler {
 
-  @Override
-  public Object execute(ExecutionEvent event) throws ExecutionException {
+  @Inject private ConnectionHandler connectionHandler;
+  @Inject private ISarosSessionManager sessionManager;
+
+  public ShareResourcesHandler() {
+    SarosPluginContext.initComponent(this);
+  }
+
+  @Execute
+  public Object execute() {
     WizardUtils.openStartSessionWizard(
         SelectionRetrieverFactory.getSelectionRetriever(IResource.class).getSelection());
     return null;
   }
+
+  @CanExecute
+  public boolean canExecute() {
```

Comment:
I actually find this canExecute method really confusing. The visibleWhen checks are still in the plugin.xml file. It seems this is the enabledWhen check. However this method is different from SessionAddSelectedResourcesHandler class. So why is there no check if the resource is already shared ? And last but not least looking at the SessionAddSelectedResourcesHandler class, how does the method know how it has to convert the selections ?

Reply:
The plugin.xml still contains the popup menus which have not yet been migrated, thats where the visibleWhen expression are from atm.
ShareResourceHandler and SessionAddSelectedResourcesHandler are fundamentally different, one starts a session the other one adds to an existing session.
I think youre right regarding the SessionAddSelectedResourcesHandler @CanExecute Method that it cant convert the Selection and the Button in the Popup Menu is not yet working properly.

Reply:
I just looked at the old plugin.xml code and it seems this handler is a bit "misused" through the magic that happens on line 53 which makes it really hard to understand.

So is the design of e4 where the canExecute method gets the selection data but when the framework calls the execute method you have to query the selection service on your own ?